### PR TITLE
Run regen-docs for pytest 3.0

### PR DESCRIPTION
**Pull request by nicoddemus**

### CODE REVIEW

File: doc/en/assert.rst [hunk 1]
```diff
@@ -246,8 +246,7 @@ the conftest file::
            f1 = Foo(1)
            f2 = Foo(2)
    >       assert f1 == f2
-   E       assert Comparing Foo instances:
-   E            vals: 1 != 2
+   E       AssertionError
```
Comment (The-Compiler) [inline doc/en/assert.rst]: Hmm, this seems like a weird change seeing that the doc is about `pytest_assertrepr_compare`?

File: doc/en/usage.rst [hunk 2]
```diff
@@ -326,5 +326,7 @@ hook was invoked::
     $ python myinvoke.py
     *** test run reporting finishing
     
+    ======= pytest-warning summary ========
+    WC1 None passing a string to pytest.main() is deprecated, pass a list of arguments instead.
```
Comment (The-Compiler) [inline doc/en/usage.rst]: I guess we should be a good example and not pass a string here :wink:

File: doc/en/assert.rst (hunk 1 above)
Comment (nicoddemus) [inline doc/en/assert.rst]: Hmmm yes, and I think that's a regression :grimacing:

File: doc/en/usage.rst (hunk 2 above)
Comment (nicoddemus) [inline doc/en/usage.rst]: Fixed, thanks! :wink:

File: doc/en/assert.rst (hunk 1 above)
Comment (nicoddemus) [inline doc/en/assert.rst]: Opened #1784 to investigate this.

File: doc/en/cache.rst [hunk 3]
```diff
@@ -226,23 +226,16 @@ You can always peek at the content of the cache using the
...
-    test_caching.py:14: AssertionError
-    ======= 1 failed in 0.12 seconds ========
+    cachedir: $REGENDOC_TMPDIR/.cache
+    ------------------------------- cache values -------------------------------
+    cache/lastfailed contains:
+      {'test_caching.py::test_function': True}
+    example/value contains:
+      42
+    
+    ======= no tests ran in 0.12 seconds ========
```
Comment (flub) [inline doc/en/cache.rst]: Is this meant to no longer run tests?

File: doc/en/example/parametrize.rst [hunk 4]
```diff
@@ -397,8 +397,11 @@ is to be run with different sets of arguments for its three arguments:
 Running it results in some skips if we don't have all the python interpreters installed and otherwise runs all combinations (5 interpreters times 5 interpreters times 3 objects to serialize/deserialize)::
 
    . $ pytest -rs -q multipython.py
-   ...........................
-   27 passed in 0.12 seconds
+   ssssssssssss...ssssssssssss
+   ======= short test summary info ========
+   SKIP [12] $REGENDOC_TMPDIR/CWD/multipython.py:23: 'python3.3' not found
+   SKIP [12] $REGENDOC_TMPDIR/CWD/multipython.py:23: 'python2.6' not found
+   3 passed, 24 skipped in 0.12 seconds
```
Comment (flub) [inline doc/en/example/parametrize.rst]: You might have been missing some interpreters when re-building this...

Comment (nicoddemus) [inline doc/en/example/parametrize.rst]: Yes, but I think that's fine to show case what happens when missing some interepreters. :grin:

File: doc/en/example/reportingdemo.rst [hunk 5]
```diff
@@ -482,8 +482,9 @@ get on the terminal - we are working on that):
             s = "123"
             g = "456"
     >       assert s.startswith(g)
-    E       assert <built-in method startswith of str object at 0xdeadbeef>('456')
-    E        +  where <built-in method startswith of str object at 0xdeadbeef> = '123'.startswith
+    E       assert False
```
Comment (flub) [inline doc/en/example/reportingdemo.rst]: Is this a regression?  I seem to remember code in rewrite to strip out this `False` bit.

Comment (nicoddemus) [inline doc/en/example/reportingdemo.rst]: I think we actually reintroduced showing `assert False` always

Comment (nicoddemus) [inline doc/en/example/reportingdemo.rst]: Yep: #1654

File: doc/en/cache.rst (hunk 3 above)
Comment (nicoddemus) [inline doc/en/cache.rst]: I think so, at least that's what the `cacheshow` function does.

File: doc/en/assert.rst (hunk 1 above)
Comment (RonnyPfannschmidt) [inline doc/en/assert.rst]: should we still merge?

Comment (The-Compiler) [inline doc/en/assert.rst]: I'll let @nicoddemus decide - I think we should probably fix #1784 before releasing 3.0, right? If so, we'd need to re-run regendoc anyways, so not much point in merging then from my POV.

Comment (flub) [inline doc/en/assert.rst]: next merge will be smaller :)

Comment (nicoddemus) [inline doc/en/assert.rst]: :+1: Thanks!

**Pull request merged**